code_generation/producer.py

```diff
@@ -45,6 +46,13 @@ def __init__(self, name, call, input, output, scopes):
         log.debug("| scopes: {}".format(self.scopes))
         log.debug("-----------------------------------------")
 
+    def check_output(self):
+        if self.output != None and not self.output_checked:
+            for scope in self.scopes:
+                for output_quantity in self.output:
+                    output_quantity.check_scope(scope)
```

Comment:
better call the method 'reserve'. That it is scope specific is clear by the argument.